src/plugins/part.c

```diff
@@ -278,16 +278,52 @@ static void close_context (struct fdisk_context *cxt) {
     fdisk_unref_context (cxt);
 }
 
-static gboolean write_label (struct fdisk_context *cxt, const gchar *disk, GError **error) {
+static gboolean write_label (struct fdisk_context *cxt, struct fdisk_table *orig, const gchar *disk, GError **error) {
     gint ret = 0;
-    ret = fdisk_write_disklabel (cxt);
+    gint dev_fd = 0;
+    guint num_tries = 1;
+
+    /* XXX: try to grab a lock for the device so that udev doesn't step in
+       between the two operations we need to perform (see below) with its
+       BLKRRPART ioctl() call which makes the device busy */
```

Comment:
Link this:
https://systemd.io/BLOCK_DEVICE_LOCKING/

Reply:
Completely missed the code was using LOCK_SH. Thanks for fixing it.

Reply:
The `LOCK_SH` is copy-pasted from our existing libparted based implementation. Not sure why we are not using the `LOCK_EX`, everything seems to be working correctly with it.